src/components/v5/shared/CopyTooltip/CopyTooltip.tsx

```diff
@@ -0,0 +1,143 @@
+import { Check } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import React, {
+  type FC,
+  type PropsWithChildren,
+  useEffect,
+  useState,
+} from 'react';
+import { usePopperTooltip } from 'react-popper-tooltip';
+
+import tooltipClasses from './CopyTooltip.styles.ts';
+import { type CopyTooltipProps } from './types.ts';
+
+const displayName = 'v5.CopyTooltip';
+
+const CopyTooltip: FC<PropsWithChildren<CopyTooltipProps>> = ({
+  children,
+  tooltipContent,
+  placement = 'auto',
+  popperOptions,
+  interactive,
+  offset = [0, 12],
+  showArrow = true,
+  isOpen,
+  isSuccess = false,
+  isFullWidthContent,
+  className,
+  selectTriggerRef = (v) => v,
+}) => {
+  const [isTooptipVisible, setIsTooltipVisible] = useState(false);
+  const {
+    getArrowProps,
+    getTooltipProps,
+    setTooltipRef,
+    setTriggerRef,
+    tooltipRef,
+    triggerRef,
+    visible,
+  } = usePopperTooltip(
+    {
+      delayShow: 200,
+      placement,
+      trigger: 'click',
+      visible: isTooptipVisible,
+      delayHide: interactive ? 200 : 0,
+      interactive,
+      offset,
+    },
+    popperOptions,
+  );
+
+  useEffect(() => {
+    if (isOpen !== undefined) {
+      setIsTooltipVisible(isOpen);
+    }
+  }, [isOpen, setIsTooltipVisible]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const isOutsideTooltip =
+        tooltipRef && !tooltipRef.contains(event.target as Node);
+
+      if (isOutsideTooltip) {
+        setIsTooltipVisible(false);
+      }
+    };
+
+    const handleMouseLeave = (event: MouseEvent) => {
+      const isOutsideTrigger =
+        triggerRef && !triggerRef.contains(event.target as Node);
+
+      if (isOutsideTrigger) {
+        setIsTooltipVisible(false);
+      }
+    };
+
+    if (visible) {
+      document.addEventListener('click', handleClickOutside);
+      document.addEventListener('mousemove', handleMouseLeave);
+    } else {
+      document.removeEventListener('click', handleClickOutside);
+      document.removeEventListener('mousemove', handleMouseLeave);
+    }
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+      document.removeEventListener('mousemove', handleMouseLeave);
+    };
+  }, [tooltipRef, triggerRef, visible, setIsTooltipVisible]);
+
+  return (
+    <>
+      <div
+        className={clsx(className, 'flex', {
+          'w-full': isFullWidthContent,
```

Comment:
i would maybe just ditch this prop since we can pass w-full in the className if need be